core/chains/evm/logpoller/log_poller.go

```diff
@@ -1103,3 +1111,75 @@ func EvmWord(i uint64) common.Hash {
 	binary.BigEndian.PutUint64(b, i)
 	return common.BytesToHash(b)
 }
+
+type notifyEntry struct {
+	index    int
+	value    common.Hash
+	notifyCh chan struct{}
+}
+
+// Notify returns a channel that emits events when a new log with the given topic key and value is saved.
+func (lp *logPoller) Notify(topicIndex int, topicValue common.Hash) <-chan struct{} {
+	if err := validateTopicIndex(topicIndex); err != nil {
+		panic(err.Error())
+	}
+
+	// Add buffer size of 1 so that if a receiver is not ready to receive, it will
+	// still get a notification of updates when it becomes ready.
+	notifyCh := make(chan struct{}, 1)
+
+	lp.notifyMu.Lock()
+	defer lp.notifyMu.Unlock()
+
+	lp.notifyEntries = append(lp.notifyEntries, notifyEntry{
+		index:    topicIndex,
+		value:    topicValue,
+		notifyCh: notifyCh,
+	})
+	return notifyCh
+}
```

Comment:
Log poller was designed as a replacement for log broadcaster, with the intentional goal of simplifying the feature set down from something that was responsible for notifications and subscriptions to something much simpler and cleaner whose only responsibility is polling and responding to queries.

Mercury may be a special case, but unfortunately I can't approve this PR as it stands--it adds back in a feature (notifications) which takes it back outside of these design goals.  We have product managers and channels for requesting features for log poller, since its feature set is something which needs to be standard across all teams, consistent and well-documented (still working on the documentation).  It's expected to also be used by 3rd parties.

You should be able to accomplish the same thing by using event broadcaster.  This would keep the core responsibilities of log poller itself fixed, leaving the responsibility for broadcasting the notifications with event broadcaster.

You can add a new pg channel `insert_on_evm_logs` to this list of eb channels:
https://github.com/smartcontractkit/chainlink/blob/develop/core/services/pg/channels.go

And create a trigger for it with something like this:
```
CREATE FUNCTION public.notifysavedlog() RETURNS trigger
  LANGUAGE plpgsql
  AS $$
      BEGIN
	PERFORM pg_notify('insert_on_evm_logs`::text, NOW()::text);
	RETURN NULL;
      END
   $$;
 ```
Then create a listener by calling `eb.eventBroadcaster.Subscribe(pg.ChannelInsertOnEvmLogs, "")` like this:
https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/txmgr/eth_broadcaster.go#L192-L195 
 
 See also:
 https://github.com/smartcontractkit/chainlink/blob/develop/core/services/pg/event_broadcaster.go

Reply:
Thanks Domino for the review. I can take a stab at using the event broadcaster!

Reply:
cc @reductionista @samsondav Do you mind taking another look? I got this working with event broadcaster. Also, I'm not sure if there is an easier way to get an instance of the event broadcaster, but I had to pass it down all the way from the top level Application struct.